config/routes.rb

```diff
@@ -1,6 +1,10 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  get '/users', to: 'users#index'
+  get '/users/:id', to: 'users#show'
 
-  # Defines the root path route ("/")
-  # root "articles#index"
-end
+  # Route for listing all posts for a user
+  get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+
+  # Route for showing a specific post for a user
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
+end
```

Comment:
- [ ] Here it is a requirement that we DO NOT expose any unnecessary endpoints by using the `only` parameters on our routes, kindly do so.

e,g
```

# config/routes.rb

  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end

```
This way we clearly state that the only routes available are the index and post for both the `user` and the `post`